File: frontend/src/types/strategy-basket.ts
import { AssetType, Currency } from './finance';

export interface StrategyBasketItem {
  id: number;
  strategyBasketId: number;
  symbol: string;
  name: string;
  assetType: AssetType;
  currency: Currency;
  quantity: number;
  currentPrice: number;
  targetPrice: number;
  targetProfitPercent: number;
  platform?: string;
  notes?: string;
  requiredCapital: number;
  targetValue: number;
  targetProfitLoss: number;
  calculatedProfitPercent: number;
}

export interface StrategyBasket {
  id: number;
  name: string;
  description?: string;
  targetDate?: string;
  isArchived: boolean;
  createdAt: string;
  items: StrategyBasketItem[];
  totalRequiredCapitalTRY: number;
  totalTargetValueTRY: number;
  totalTargetProfitLossTRY: number;
  totalTargetProfitPercentTRY: number;
  totalRequiredCapitalUSD: number;
  totalTargetValueUSD: number;
  totalTargetProfitLossUSD: number;
}

File: backend/SpendLogV2.Domain/Entities/StrategyBasket.cs
using SpendLogV2.Domain.Common;

namespace SpendLogV2.Domain.Entities;

public class StrategyBasket : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? TargetDate { get; set; }
    public bool IsArchived { get; set; } = false;

    public ICollection<StrategyBasketItem> Items { get; set; } = new List<StrategyBasketItem>();
}
